File: Timetable.App/src/app/_models/lesson.ts
export class Lesson {
    id: string;
    name: string;
    start: Date;
    duration: number;
    classroom: string;
    link: string;
    groupId: number;

    hexColor?: string;
    isVisible: boolean = true;

    constructor(data: any)
    {
        this.id = data.id;
        this.name = data.name;
        this.start = new Date(data.start);
        this.duration = data.duration;
        this.classroom = data.classroom;
        this.link = data.link;
        this.groupId = data.groupId;
    }
}

File: TimetableA.DataAccessLayer.RavenDB/Models/Lesson.cs
namespace TimetableA.DataAccessLayer.RavenDB.Models;

internal class Lesson : IDocument
{
    public string? Id { get; set; }

    public Lesson() { }

    public Lesson(int duration)
    {
        Duration = TimeSpan.FromMinutes(duration);
    }

    public string Name { get; set; }

    public DateTime Start { get; set; }

    public TimeSpan Duration { get; set; }

    public string Classroom { get; set; }

    public string Link { get; set; }

    public string? GroupId { get; set; }
}
